src/NumericalAlgorithms/LinearOperators/ExponentialFilter.hpp

```diff
@@ -112,6 +129,7 @@ class Exponential {
   double alpha_{36.0};
   unsigned half_power_{16};
   bool enable_{true};
+  std::optional<std::vector<std::string>> blocks_to_filter_{};
```

Comment:
would storing this as a `std::opt<unordered_set>` make lookup easier? Doing the `any_of` is a linear search, which is less ideal than a hash, I think. No big deal either way, I think, though.